recipes/tcl_installer/8.6.10/conanfile.py

```diff
@@ -0,0 +1,202 @@
+        for root, _, files in os.walk(self.build_folder):
+            if "Makefile" in files:
+                tools.replace_in_file(os.path.join(root, "Makefile"), "-Dstrtod=fixstrtod", "", strict=False)
+
+        with tools.chdir(self.build_folder):
+            autoTools.make()
+
+    def build(self):
+        self._fix_sources()
+        if self.settings.compiler == "Visual Studio":
+            # do not treat warnings as errors 
+            tools.replace_in_file(os.path.join(self._source_subfolder, "win", "rules.vc"), "cwarn = $(cwarn) -WX", "")
+            self._build_nmake(["release"])
+        else:
+            self._build_autotools()
+
+    def package(self):
+        if self.settings.compiler == "Visual Studio":
+            self._build_nmake(["install-binaries", "install-libraries"])
+            # there is no way to only install headers + libraries
+            # hence why we build/install everything and then remove
+        else:
+            with tools.chdir(self.build_folder):
+                autoTools = self._get_auto_tools()
+                autoTools.make(target="install-binaries")
+                autoTools.make(target="install-libraries")
+                autoTools.make(target="install-msgs")
+                autoTools.make(target="install-tzdata")
+                autoTools.make(target="install-headers")
+                autoTools.make(target="install-private-headers")
+            pkgconfig_dir = os.path.join(self.package_folder, "lib", "pkgconfig")
+            tools.rmdir(pkgconfig_dir)
+        self.copy(pattern="license.terms", dst="licenses", src=self._source_subfolder)
+
+        tclConfigShPath = os.path.join(self.package_folder, "lib", "tclConfig.sh")
+        package_path = os.path.join(self.package_folder)
+        if self._is_mingw_windows:
+            package_path = package_path.replace("\\", "/")
+        tools.replace_in_file(tclConfigShPath,
+                              package_path,
+                              "${TCL_ROOT}")
+        tools.replace_in_file(tclConfigShPath,
+                              "\nTCL_BUILD_",
+                              "\n#TCL_BUILD_")
+
+        tools.replace_in_file(tclConfigShPath,
+                              "\nTCL_SRC_DIR",
+                              "\n#TCL_SRC_DIR")
+
+    def package_info(self):
+        libdirs = []
+        for root, _, _ in os.walk(os.path.join(self.package_folder, "lib"), topdown=False):
+            newlibs = tools.collect_libs(self, root)
+            if newlibs:
+                libdirs.append(root)
+
+        self.cpp_info.bindirs = ["bin"]
+        self.cpp_info.libdirs = libdirs
+        self.cpp_info.includedirs = ["include"]
+
+        tcl_library = os.path.join(self.package_folder, "lib", "{}{}".format(self.name, ".".join(self.version.split(".")[:2])))
+        self.output.info("Setting TCL_LIBRARY environment variable to {}".format(tcl_library))
+        self.env_info.TCL_LIBRARY = tcl_library
```

Comment:
These variables will clash with the current `tcl`.
Or in the future, `tcl` and `libtcl`.